pkg/cmd/browse/browse_test.go

```diff
@@ -45,37 +136,87 @@ func Test_runBrowse(t *testing.T) {
 			name: "no arguments",
 			opts: BrowseOptions{
 				SelectorArg: "",
-				FlagAmount:  0,
 			},
-			baseRepo:    ghrepo.New("jessica", "cli"),
-			expectedURL: "https://github.com/jessica/cli",
+			baseRepo:      ghrepo.New("jessica", "cli"),
+			defaultBranch: "trunk",
+			expectedURL:   "https://github.com/jessica/cli/tree/trunk/",
 		},
 		{
 			name:          "file argument",
 			opts:          BrowseOptions{SelectorArg: "path/to/file.txt"},
-			baseRepo:      ghrepo.New("bchadwic", "cli"),
+			baseRepo:      ghrepo.New("ken", "cli"),
 			defaultBranch: "main",
-			expectedURL:   "https://github.com/bchadwic/cli/tree/main/path/to/file.txt",
+			expectedURL:   "https://github.com/ken/cli/tree/main/path/to/file.txt",
 		},
 		{
 			name: "branch flag",
 			opts: BrowseOptions{
-				Branch:     "trunk",
-				FlagAmount: 1,
+				Branch: "trunk",
 			},
-			baseRepo:    ghrepo.New("bchadwic", "cli"),
-			expectedURL: "https://github.com/bchadwic/cli/tree/trunk",
+			baseRepo:    ghrepo.New("thanh", "cli"),
+			expectedURL: "https://github.com/thanh/cli/tree/trunk/",
 		},
 		{
 			name: "settings flag",
 			opts: BrowseOptions{
 				SettingsFlag: true,
-				FlagAmount:   1,
 			},
 			baseRepo:    ghrepo.New("bchadwic", "cli"),
 			expectedURL: "https://github.com/bchadwic/cli/settings",
 		},
+		{
+			name: "projects flag",
+			opts: BrowseOptions{
+				ProjectsFlag: true,
+			},
+			baseRepo:    ghrepo.New("bchadwic", "cli"),
+			expectedURL: "https://github.com/bchadwic/cli/projects",
+		},
+		{
+			name: "wiki flag",
+			opts: BrowseOptions{
+				WikiFlag: true,
+			},
+			baseRepo:    ghrepo.New("bchadwic", "cli"),
+			expectedURL: "https://github.com/bchadwic/cli/wiki",
+		},
+		{
+			name: "issue argument",
+			opts: BrowseOptions{
+				SelectorArg: "217",
+			},
+			baseRepo:    ghrepo.New("bchadwic", "cli"),
+			expectedURL: "https://github.com/bchadwic/cli/issues/217",
+		},
+		{
+			name: "file with line number",
+			opts: BrowseOptions{
+				SelectorArg: "path/to/file.txt:32",
+			},
+			baseRepo:      ghrepo.New("bchadwic", "cli"),
+			defaultBranch: "trunk",
+			expectedURL:   "https://github.com/bchadwic/cli/tree/trunk/path/to/file.txt#L32",
+		},
+		{
+			name: "file with invalid line number",
+			opts: BrowseOptions{
+				SelectorArg: "path/to/file.txt:32:32",
+			},
+			baseRepo:      ghrepo.New("bchadwic", "cli"),
+			defaultBranch: "trunk",
```

Comment:
Minor: could we remove the `defaultBranch` stub from this test to ensure that this invocation does not consult the API?

Ideally, since the user has supplied invalid arguments, we should error out before doing any API lookups. If a lookup is being done, but we error out later due to input, please consider erroring out earlier.